facade/src/main/java/org/jboss/pnc/facade/providers/SCMRepositoryProviderImpl.java

```diff
@@ -160,7 +162,16 @@ public RepositoryCreationResponse createSCMRepository(
                         + preBuildSyncEnabled + ")");
         if (StringUtils.isEmpty(scmUrl))
             throw new InvalidEntityException("You must specify the SCM URL.");
-
+        if (scmUrl.startsWith("http") || scmUrl.startsWith("https")) {
+            try {
+                URL url = new URL(scmUrl);
+                if (url.getProtocol().equals("http") || url.getProtocol().equals("https")) {
+                    scmUrl = "git+http://" + url.getHost() + url.getFile();
```

Comment:
In some specific cases, you may want to also add the port if specified in the url.